src/Microsoft.AspNetCore.Mvc.RazorPages/IPageActivatorProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages
+{
+    /// <summary>
+    /// Provides methods to create a Razor page.
+    /// </summary>
+    public interface IPageActivatorProvider
+    {
+        /// <summary>
+        /// Creates a Razor page activator.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="CompiledPageActionDescriptor"/>.</param>
+        Func<PageContext, object> Create(CompiledPageActionDescriptor descriptor);
+
+        /// <summary>
+        /// Releases a Razor page.
+        /// </summary>
+        /// <param name="context">The <see cref="PageContext"/>.</param>
+        /// <param name="page">The page to release.</param>
+        void Release(PageContext context, object page);
```

Comment:
Suggestions for what this needs to look like? Should `Create` return a pair (or an interface) of creators of destroyers? Or should this be a separate `Action<PageContext, object>` that we independently cache?